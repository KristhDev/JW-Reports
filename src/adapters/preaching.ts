import { Preaching, PreachingEndpoint } from '../interfaces';

/**
 * Creates a new Preaching object from a PreachingEndpoint object.
 *
 * @param {PreachingEndpoint} preaching - The PreachingEndpoint object to convert.
 * @return {Preaching} The converted Preaching object.
 */
export const preachingAdapter = (preaching: PreachingEndpoint): Preaching => ({
    id: preaching.id,
    userId: preaching.user_id,
    day: preaching.day,
    initHour: preaching.init_hour,
    finalHour: preaching.final_hour,
    createdAt: preaching.created_at,
    updatedAt: preaching.updated_at
});