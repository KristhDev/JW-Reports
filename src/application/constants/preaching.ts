import { ParticipateInMinistry } from '@infrasturcture/interfaces';

/**
 * The headers of the table that will be displayed in the PreachingTable component
 */
export const TABLE_PREACHING_HEADERS = [
    'N°',
    'Día',
    'Hora de Inicio',
    'Hora de Fin'
];

export const HOURS_REQUIREMENTS = {
    'ninguno': 0,
    'auxiliar': 30,
    'regular': 50,
    'especial': 90
}

export const preachingMessages = {
    ADDED_SUCCESS: 'Has agregado el día de predicación correctamente.',
    DAY_REQUIRED: 'El día no puede estar vacío.',
    DELETED_SUCCESS: 'Has eliminado el día de predicación correctamente.',
    FINAL_HOUR_REQUIRED: 'La hora final no puede estar vacía.',
    INIT_HOUR_GREATER_THAN_FINAL: 'La hora inicial no puede ser mayor que la hora final.',
    INIT_HOUR_REQUIRED: 'La hora inicial no puede estar vacía.',
    MONTHLY_HOURS_REQUIRED_DONE: '¡Excelente! has cumplido con tu requerimiento de horas por mes.',
    UNSELECTED_DELETE: 'No hay un día de predicación seleccionado para eliminar.',
    UNSELECTED_UPDATE: 'No hay un día de predicación seleccionado para actualizar.',
    UPDATED_SUCCESS: 'Has actualizado el día de predicación correctamente.',
    WEEKLY_HOURS_REQUIRED_DONE: '¡Excelente! has cumplido con tu requerimiento de horas por semana.',
}

export const MINISTRY_PARTICIPATIONS: { label: string, value: ParticipateInMinistry }[] = [
    { label: 'Si', value: 'si' },
    { label: 'No', value: 'no' }
];