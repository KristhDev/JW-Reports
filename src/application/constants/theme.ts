import { ItemOption } from '../../infrasturcture/interfaces';

/**
 * The theme options to select
 */
export const THEME_OPTIONS: ItemOption[] = [
    { label: 'Modo predeterminado', value: 'default' },
    { label: 'Modo claro', value: 'light' },
    { label: 'Modo oscuro', value: 'dark' }
];