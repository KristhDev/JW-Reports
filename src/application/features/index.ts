export * from './auth';
export * from './courses';
export * from './lessons';
export * from './permissions';
export * from './preaching';
export * from './revisits';
export * from './status';
export * from './types';
export * from './ui';