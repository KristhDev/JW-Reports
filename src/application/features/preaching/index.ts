export {
    default as preachingReducer,
    INIT_PREACHING,
    PREACHING_INITIAL_STATE,
    addPreaching,
    clearPreaching,
    removePreaching,
    setIsPreachingDeleting,
    setIsPreachingLoading,
    setIsPreachingsLoading,
    setPreachings,
    setSelectedDate,
    setSelectedPreaching,
    updatePreaching
} from './slice';

export * from './types';