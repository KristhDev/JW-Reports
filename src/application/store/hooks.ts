import { useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();