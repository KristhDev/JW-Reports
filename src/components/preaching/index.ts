export { PreachingForm } from './PreachingForm';
export { PreachingTable } from './PreachingTable';
export { RevisitCard } from './RevisitCard';