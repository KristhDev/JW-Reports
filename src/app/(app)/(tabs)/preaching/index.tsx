import { Redirect } from 'expo-router';

/* Constants */
import { precursors } from '@application/constants';

/* Auth */
import { useAuth } from '@auth';

/**
 * Redirects to either the Precursor or Publisher tab based on the user's
 * assigned role.
 *
 * @returns A redirect to either the Precursor or Publisher tab.
 */
export default function PreachingRootScreen(): JSX.Element {
    const { state: { user } } = useAuth();

    if (user.precursor !== precursors.NINGUNO) return <Redirect href="/(app)/(tabs)/preaching/precursor" />
    else return <Redirect href="/(app)/(tabs)/preaching/publisher" />
}