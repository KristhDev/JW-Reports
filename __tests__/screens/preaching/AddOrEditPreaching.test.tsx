import React from 'react';
import { act, render, screen, waitFor } from '@testing-library/react-native';

/* Screens */
import { AddOrEditPreaching } from '../../../src/screens/preaching';

/* Features */
import { INIT_PREACHING } from '../../../src/features/preaching';
import { preachingsState } from '../../features/preaching';

/* Hooks */
import { usePreaching, useStatus, useTheme } from '../../../src/hooks';

/* Theme */
import { darkColors } from '../../../src/theme';

/* Mock hooks */
jest.mock('../../../src/hooks/usePreaching.ts');
jest.mock('../../../src/hooks/useStatus.ts');
jest.mock('../../../src/hooks/useTheme.ts');

describe('Test in <AddOrEditPreaching /> screen', () => {
    (usePreaching as jest.Mock).mockReturnValue({
        state: {
            ...preachingsState,
            seletedPreaching: {
                ...INIT_PREACHING,
                day: '2022-12-29 00:00:00',
                init_hour: '2022-12-30 09:00:00',
                final_hour: '2022-12-30 09:00:00'
            }
        },
        savePreaching: jest.fn(),
        updatePreaching: jest.fn()
    });

    (useStatus as jest.Mock).mockReturnValue({
        setErrorForm: jest.fn()
    });

    (useTheme as jest.Mock).mockReturnValue({
        state: { colors: darkColors },
    });

    it('should to match snapshot', async () => {
        await waitFor(() => {
            render(<AddOrEditPreaching />);
        });

        await act(async () => {
            await waitFor(() => {
                expect(screen.toJSON()).toMatchSnapshot();
            });
        });
    });

    it('should render respective title when seletedPreaching is empty', async () => {
        await waitFor(() => {
            render(<AddOrEditPreaching />);
        });

        await act(async () => {
            await waitFor(() => {

                /* Get title */
                const title = screen.getByTestId('title-text');

                /* Check if title exists and contain value pass by props */
                expect(title).toBeTruthy();
                expect(title.props.children).toBe('Agregar día de predicación');
            });
        })
    });

    it('should render respective title when seletedPreaching isnt empty', async () => {

        /* Mock data in usePreaching */
        (usePreaching as jest.Mock).mockReturnValue({
            state: {
                ...preachingsState,
                seletedPreaching: {
                    ...INIT_PREACHING,
                    id: '0c0d257e-5031-4111-9f0b-d10cdb2efb92',
                    user_id: 'f366c803-37cc-43c9-91c1-c78ec6a0718d',
                    day: '2022-12-29 00:00:00',
                    init_hour: '2022-12-30 09:00:00',
                    final_hour: '2022-12-30 10:30:00'
                }
            },
            savePreaching: jest.fn(),
            updatePreaching: jest.fn()
        });

        await waitFor(() => {
            render(<AddOrEditPreaching />);
        });

        await act(async () => {
            await waitFor(() => {

                /* Get title */
                const title = screen.getByTestId('title-text');

                /* Check if title exists and contain value pass by props */
                expect(title).toBeTruthy();
                expect(title.props.children).toBe('Editar día de predicación');
            });
        });
    });
});