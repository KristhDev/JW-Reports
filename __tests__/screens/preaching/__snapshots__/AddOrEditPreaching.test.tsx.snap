// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <AddOrEditPreaching /> screen should to match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "padding": 20,
          "width": "100%",
        },
        Object {
          "paddingBottom": 0,
          "paddingTop": 30,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 38,
            "fontWeight": "bold",
          },
          Object {
            "fontSize": 24,
          },
        ]
      }
      testID="title-text"
    >
      Agregar día de predicación
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingBottom": 40,
        "paddingTop": 30,
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "padding": 10,
          "width": 675,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "marginBottom": 5,
              "padding": 5,
            },
            undefined,
          ]
        }
        testID="datetimefield-label"
      >
        Día de predicación:
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "#FFFFFF",
                "borderRadius": 5,
                "borderWidth": 0.5,
                "flex": 1,
                "flexDirection": "row",
              },
              undefined,
            ]
          }
        >
          <TextInput
            autoCorrect={false}
            editable={false}
            modalTitle="Día de predicación"
            placeholder="Seleccione el día"
            placeholderTextColor="#BFBFBF"
            selectionColor="#A0B9E2"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 17,
                  "paddingLeft": 10,
                },
                undefined,
              ]
            }
            testID="datetimefield-text-input"
            value="29"
          />
        </View>
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              false,
              Array [
                Object {
                  "backgroundColor": "#C0A7E1",
                  "borderRadius": 10,
                },
                Object {
                  "marginLeft": 10,
                },
              ],
            ]
          }
          testID="button-touchable"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingHorizontal": 20,
                  "paddingVertical": 8,
                },
                Object {
                  "paddingHorizontal": 10,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "color": "#121212",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "marginBottom": 2,
                      "textAlign": "center",
                    },
                    undefined,
                  ],
                ]
              }
              testID="button-text"
            />
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#121212",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <Modal
        animationType="none"
        hardwareAccelerated={false}
        isVisible={false}
        onHide={[Function]}
        transparent={true}
        visible={false}
      />
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "padding": 10,
          "width": 675,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "marginBottom": 5,
              "padding": 5,
            },
            undefined,
          ]
        }
        testID="form-field-label"
      >
        Publicaciones:
      </Text>
      <View
        style={
          Object {
            "borderColor": "transparent",
            "borderRadius": 7,
            "borderWidth": 1,
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "#FFFFFF",
              "borderRadius": 5,
              "borderWidth": 0.5,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "transparent",
                  "borderRadius": 5,
                  "borderWidth": 1.5,
                  "flexDirection": "row",
                  "paddingRight": 10,
                },
                undefined,
              ]
            }
          >
            <TextInput
              autoCorrect={false}
              cursorColor="#C0A7E1"
              keyboardType="numeric"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSelectionChange={[Function]}
              placeholder="Número de publicaciones"
              placeholderTextColor="#BFBFBF"
              selection={
                Object {
                  "end": 1,
                  "start": 1,
                }
              }
              selectionColor="#A0B9E2"
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "flex": 1,
                    "fontSize": 17,
                    "paddingLeft": 10,
                    "paddingRight": 5,
                    "textAlignVertical": "center",
                  },
                  undefined,
                ]
              }
              testID="form-field-text-input"
              value="0"
            />
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#BFBFBF",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "padding": 10,
          "width": 675,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "marginBottom": 5,
              "padding": 5,
            },
            undefined,
          ]
        }
        testID="form-field-label"
      >
        Videos:
      </Text>
      <View
        style={
          Object {
            "borderColor": "transparent",
            "borderRadius": 7,
            "borderWidth": 1,
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "#FFFFFF",
              "borderRadius": 5,
              "borderWidth": 0.5,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "transparent",
                  "borderRadius": 5,
                  "borderWidth": 1.5,
                  "flexDirection": "row",
                  "paddingRight": 10,
                },
                undefined,
              ]
            }
          >
            <TextInput
              autoCorrect={false}
              cursorColor="#C0A7E1"
              keyboardType="numeric"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSelectionChange={[Function]}
              placeholder="Número de videos"
              placeholderTextColor="#BFBFBF"
              selection={
                Object {
                  "end": 1,
                  "start": 1,
                }
              }
              selectionColor="#A0B9E2"
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "flex": 1,
                    "fontSize": 17,
                    "paddingLeft": 10,
                    "paddingRight": 5,
                    "textAlignVertical": "center",
                  },
                  undefined,
                ]
              }
              testID="form-field-text-input"
              value="0"
            />
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#BFBFBF",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "padding": 10,
          "width": 675,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "marginBottom": 5,
              "padding": 5,
            },
            undefined,
          ]
        }
        testID="form-field-label"
      >
        Revisitas:
      </Text>
      <View
        style={
          Object {
            "borderColor": "transparent",
            "borderRadius": 7,
            "borderWidth": 1,
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "#FFFFFF",
              "borderRadius": 5,
              "borderWidth": 0.5,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderColor": "transparent",
                  "borderRadius": 5,
                  "borderWidth": 1.5,
                  "flexDirection": "row",
                  "paddingRight": 10,
                },
                undefined,
              ]
            }
          >
            <TextInput
              autoCorrect={false}
              cursorColor="#C0A7E1"
              keyboardType="numeric"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSelectionChange={[Function]}
              placeholder="Número de revisitas"
              placeholderTextColor="#BFBFBF"
              selection={
                Object {
                  "end": 1,
                  "start": 1,
                }
              }
              selectionColor="#A0B9E2"
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "flex": 1,
                    "fontSize": 17,
                    "paddingLeft": 10,
                    "paddingRight": 5,
                    "textAlignVertical": "center",
                  },
                  undefined,
                ]
              }
              testID="form-field-text-input"
              value="0"
            />
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#BFBFBF",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "padding": 10,
          "width": 675,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "marginBottom": 5,
              "padding": 5,
            },
            undefined,
          ]
        }
        testID="datetimefield-label"
      >
        Hora de inicio:
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "#FFFFFF",
                "borderRadius": 5,
                "borderWidth": 0.5,
                "flex": 1,
                "flexDirection": "row",
              },
              undefined,
            ]
          }
        >
          <TextInput
            autoCorrect={false}
            editable={false}
            modalTitle="Hora de inicio"
            placeholder="Seleccione la hora"
            placeholderTextColor="#BFBFBF"
            selectionColor="#A0B9E2"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 17,
                  "paddingLeft": 10,
                },
                undefined,
              ]
            }
            testID="datetimefield-text-input"
            value="09:00"
          />
        </View>
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              false,
              Array [
                Object {
                  "backgroundColor": "#C0A7E1",
                  "borderRadius": 10,
                },
                Object {
                  "marginLeft": 10,
                },
              ],
            ]
          }
          testID="button-touchable"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingHorizontal": 20,
                  "paddingVertical": 8,
                },
                Object {
                  "paddingHorizontal": 10,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "color": "#121212",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "marginBottom": 2,
                      "textAlign": "center",
                    },
                    undefined,
                  ],
                ]
              }
              testID="button-text"
            />
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#121212",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <Modal
        animationType="none"
        hardwareAccelerated={false}
        isVisible={false}
        onHide={[Function]}
        transparent={true}
        visible={false}
      />
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "padding": 10,
          "width": 675,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "marginBottom": 5,
              "padding": 5,
            },
            undefined,
          ]
        }
        testID="datetimefield-label"
      >
        Hora de fin:
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderColor": "#FFFFFF",
                "borderRadius": 5,
                "borderWidth": 0.5,
                "flex": 1,
                "flexDirection": "row",
              },
              undefined,
            ]
          }
        >
          <TextInput
            autoCorrect={false}
            editable={false}
            modalTitle="Hora de fin"
            placeholder="Seleccione la hora"
            placeholderTextColor="#BFBFBF"
            selectionColor="#A0B9E2"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 17,
                  "paddingLeft": 10,
                },
                undefined,
              ]
            }
            testID="datetimefield-text-input"
            value="09:00"
          />
        </View>
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              false,
              Array [
                Object {
                  "backgroundColor": "#C0A7E1",
                  "borderRadius": 10,
                },
                Object {
                  "marginLeft": 10,
                },
              ],
            ]
          }
          testID="button-touchable"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingHorizontal": 20,
                  "paddingVertical": 8,
                },
                Object {
                  "paddingHorizontal": 10,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "color": "#121212",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "marginBottom": 2,
                      "textAlign": "center",
                    },
                    undefined,
                  ],
                ]
              }
              testID="button-text"
            />
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#121212",
                    "fontSize": 25,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <Modal
        animationType="none"
        hardwareAccelerated={false}
        isVisible={false}
        onHide={[Function]}
        transparent={true}
        visible={false}
      />
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          false,
          Array [
            Object {
              "backgroundColor": "#C0A7E1",
              "borderRadius": 10,
            },
            Object {
              "marginTop": 30,
            },
          ],
        ]
      }
      testID="button-touchable"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 20,
              "paddingVertical": 8,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              Object {
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "color": "#121212",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 2,
                  "textAlign": "center",
                },
                undefined,
              ],
            ]
          }
          testID="button-text"
        >
          Guardar
        </Text>
      </View>
    </View>
  </View>
</View>
`;
